modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/description/AbstractDescriptionDeclarer.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.capability.xml.description;
+
+import org.mule.runtime.extension.api.annotation.Alias;
+import org.mule.runtime.module.extension.internal.capability.xml.schema.ExtensionAnnotationProcessor;
+
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+
+/**
+ * Generic contract for classes that adds descriptions to a declaration using a {@link ProcessingEnvironment} to
+ * enrich the descriptions with the javadocs extracted from the extension's acting classes.
+ * <p>
+ * This is necessary because such documentation is not available on runtime.
+ *
+ * @param <T> the type to document.
+ * @since 4.0
+ */
+abstract class AbstractDescriptionDeclarer<T> {
```

Comment:
Why is this a Declarer? Is this class meant to be the base class of `org.mule.runtime.api.meta.model.declaration.fluent.Declarer` implementations?